src/hist/core.py

```diff
@@ -11,19 +11,26 @@
 class BaseHist(Histogram):
     def __init__(self, *args, **kwargs):
         """
-            Initialize Hist object. Axis params must contain the names.
+            Initialize BaseHist object. Axis params must contain the names.
         """
 
         super().__init__(*args, **kwargs)
         self.names: dict = dict()
         for ax in self.axes:
             if ax.name in self.names:
                 raise Exception(
-                    "Hist instance cannot contain axes with duplicated names."
+                    "BaseHist instance cannot contain axes with duplicated names."
```

Comment:
```suggestion
                    f"{self.__class__.__name__} instance cannot contain axes with duplicated names."
```

Reply:
You can use `type(self).__name__` if you like. In Python 3, `type(self) == self.__class__` always, and it is mostly always true in Python 2 (as long as you use new-style classes, which everyone should). I use the syntax above because it works in the old style `"{self.__class__.__name__}".format(self=self)` syntax, whereas calling a function like type does not. f-strings don't care. They are also faster.